test/e2e/quick_start_test.go

```diff
@@ -181,3 +181,38 @@ var _ = Describe("When following the Cluster API quick-start with dualstack and
 		}
 	})
 })
+
+var _ = Describe("When following the Cluster API quick-start check finalizers resilience after deletion", func() {
+	QuickStartSpec(ctx, func() QuickStartSpecInput {
+		return QuickStartSpecInput{
+			E2EConfig:              e2eConfig,
+			ClusterctlConfigPath:   clusterctlConfigPath,
+			BootstrapClusterProxy:  bootstrapClusterProxy,
+			ArtifactFolder:         artifactFolder,
+			SkipCleanup:            skipCleanup,
+			InfrastructureProvider: pointer.String("docker"),
+			PostMachinesProvisioned: func(proxy framework.ClusterProxy, namespace, clusterName string) {
+				// This check ensures that finalizers are resilient - i.e. correctly re-reconciled - when removed.
+				framework.ValidateFinalizersResilience(ctx, proxy, namespace, clusterName)
+			},
+		}
+	})
+})
+
+var _ = Describe("When following the Cluster API quick-start with ClusterClass check finalizers resilience after deletion [PR-Blocking] [ClusterClass]", func() {
```

Comment:
```suggestion
var _ = Describe("When following the Cluster API quick-start with ClusterClass check finalizers resilience after deletion [ClusterClass]", func() {
```
Just to remember to remove this before merging.

Reply:
I was under the impression that atleast the `ClusterClass` test should be tagged as `[PR-Blocking]` since it could be important to test Finalizers of reconciled resources? I will remove it, until it's decided.

Reply:
PR-Blocking is run on every PR so we prefer to keep it fast. There's only one PR-Blocking test on CAPI right now which is the ClusterClass quick start. 

If this test is stable enough to add to the normal quick start it will become PR Blocking.

Reply:
I introduced the `initialObjsWithFinalizers` check to ensure that we test `Finalizers` not only by resource types, but also by resource names. E.g. `MachineDeployment` (not topology owned) will never get a `Finalizer`, although the `MachineDeployment` type is on the resource map.